src/Workspaces/SharedUtilitiesAndExtensions/Compiler/VisualBasic/Utilities/CastAnalyzer.vb

```diff
@@ -132,6 +138,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
             If speculatedExpressionOuterType Is Nothing Then
                 Return Nothing
             End If
+            Stop
+            If outerSpeculatedExpression.IsParentKind(SyntaxKind.SimpleArgument) Then
+                speculatedExpressionOuterType = outerType
+                Return _semanticModel.ClassifyConversion(_castExpressionNode.WalkDownParentheses(), speculatedExpressionOuterType)
+            End If
```

Comment:
1. can we doc why we're doing this?
2. are there tests that validate hte behavior when the WalkDownParentheses is triggered?

Reply:
why does the code above use `speculationAnalyzer.SpeculativeSemanticModel` but this code directly accesses _semanticModel?

Reply:
The case that currently fails is a cast inside a SimpleArgument. The function GetSpeculatedExpressionToOuterTypeConversion gets called for SimpleArgments it seems it was written to handle a cast of a ParamArray which according to the tests that fail when I remove it should not have the cast removed. I copied the code (lines 142-145) from the If side of the caller that handles other cases.

Reply:
I don't know why speculationAnalyzer.SpeculativeSemanticModel was originally used

Reply:
WalkDownParentheses is called a lot and there any many tests that hit the inner while loop. If you are asking about  line 143 my 2 new tests don't hit the inner loop.

Reply:
can you add the tests, and use the sepculative model to match teh rest of the code around you?

Reply:
The node I need to look at at this point is not in the speculationAnalyzer.SpeculativeSemanticModel and it throws an exception. Updated test to test the while look in WalkDownParentheses.

Reply:
@CyrusNajmabadi added tests to verify line 143.

Reply:
@CyrusNajmabadi  Removed _semanticModel from line 142 section. removed most code I added to function. Special case in now in caller what already used SemanticModel.